src/base/net/downloadhandler.cpp

```diff
@@ -147,25 +147,23 @@ void Net::DownloadHandler::checkDownloadSize(qint64 bytesReceived, qint64 bytesT
 
 bool Net::DownloadHandler::saveToFile(const QByteArray &replyData, QString &filePath)
 {
-    QTemporaryFile *tmpfile = new QTemporaryFile(Utils::Fs::tempPath() + "XXXXXX");
-    if (!tmpfile->open()) {
-        delete tmpfile;
+    QTemporaryFile tmpfile {Utils::Fs::tempPath() + "XXXXXX"};
+    tmpfile.setAutoRemove(false);
+
+    if (!tmpfile.open())
         return false;
-    }
 
-    tmpfile->setAutoRemove(false);
-    filePath = tmpfile->fileName();
+    filePath = tmpfile.fileName();
     qDebug("Temporary filename is: %s", qUtf8Printable(filePath));
     if (m_reply->isOpen() || m_reply->open(QIODevice::ReadOnly)) {
-        tmpfile->write(replyData);
-        tmpfile->close();
+        tmpfile.write(replyData);
+        tmpfile.close();
         // XXX: tmpfile needs to be deleted on Windows before using the file
         // or it will complain that the file is used by another process.
```

Comment:
>Maybe QTemporaryFile was bugged back then on Windows and ::close() wasn't enough?

I've looked into the commits and it could be the case, as there are `emit ` statements there... But it is all irrelevant now, there are no `emit` and `tmpfile` will go out of scope when `return true/false` statement is executed.
Maybe I should also remove the comment?

Reply:
>Maybe QTemporaryFile was bugged back then on Windows and ::close() wasn't enough?

Maybe... But it affects only older code when we emit signal there so file can be used before QTemporaryFile is destroyed. In @Chocobo1 code QTemporaryFile  is destroyed earlier than file will be used.

Reply:
>Maybe I should also remove the comment?

I think Yes. Otherwise it confuses the reviewers.

Reply:
>I've looked into the commits and it could be the case, as there are emit statements there... But it is all irrelevant now, there are no emit and tmpfile will go out of scope when return true/false statement is executed.

Good point and you're probably correct.

>Maybe I should also remove the comment?

Yes. Since it probably isn't valid anymore.